components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -116,6 +116,117 @@ std::string URLMethodToRequestType(ledger::type::UrlMethod method) {
   }
 }
 
+bool ResetOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::vector<std::string>& paths) {
+  bool result = true;
+
+  for (const auto& path : paths) {
+    const base::FilePath file_path =
+        base::FilePath::FromUTF8Unsafe(path.c_str());
+    if (!base::DeletePathRecursively(file_path)) {
+      result = false;
+    }
+  }
+
+  if (diagnostic_log) {
+    diagnostic_log->Close();
+  }
+
+  return result;
+}
+
+bool DeleteDiagnosticLogOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::string& path) {
+  if (diagnostic_log) {
+    diagnostic_log->Close();
+  }
+
+  const base::FilePath file_path = base::FilePath::FromUTF8Unsafe(path.c_str());
+  if (!base::PathExists(file_path)) {
+    return true;
+  }
+
+  return base::DeleteFile(file_path);
+}
+
+std::string LoadDiagnosticLogOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::string& log_path,
+    const int num_lines) {
+  if (!diagnostic_log) {
+    return std::string();
+  }
+
+  const base::FilePath file_path =
+      base::FilePath::FromUTF8Unsafe(log_path.c_str());
+  if (!base::PathExists(file_path)) {
+    return std::string();
+  }
+
+  std::string value;
+  if (!TailFileAsString(diagnostic_log.get(), num_lines, &value)) {
+    return base::StringPrintf(
+        "ERROR: %s",
+        GetLastFileError(diagnostic_log.get()).c_str());
+  }
+
+  return value;
+}
+
+bool MaybeTailDiagnosticLog(
+    base::WeakPtr<base::File> diagnostic_log,
+    const int num_lines) {
+  if (!diagnostic_log || !diagnostic_log->IsValid()) {
+    return false;
+  }
+
+  static bool first_run = true;
+
+  const int64_t length = diagnostic_log->GetLength();
+  if (length == -1) {
+    return false;
+  }
+
+  if (!first_run && length <= kDiagnosticLogMaxFileSize) {
```

Comment:
Why do we not want to tail the log the first time this function is called?

Reply:
I think @tmancey may know the answer to that one.

Reply:
Added as reviewer as well.

Reply:
We do not tail the log first run so that if the browser crashes and we investigate the log with the user they are able to re-run the browser without losing past logs

Reply:
Thanks @tmancey! @emerick can you add a comment explaining `first_run`?

Reply:
Yep, good idea.